bfvmm/src/vmcs/src/vmcs_intel_x64_check_controls.cpp

```diff
@@ -361,12 +384,19 @@ vmcs_intel_x64::check_control_enable_ept_checks()
 void
 vmcs_intel_x64::check_control_enable_pml_checks()
 {
-    if (!is_enabled_pml())
+    using namespace vmcs::primary_processor_based_vm_execution_controls;
+    using namespace vmcs::secondary_processor_based_vm_execution_controls;
+
+    bool secondary_ctls = activate_secondary_controls::is_enabled();
+    bool pml = enable_pml::is_enabled();
+    bool ept = enable_ept::is_enabled();
+
+    if (!pml || !secondary_ctls)
```

Comment:
There are a couple of things to do here:

- Both PML and EPT should be `is_enabled_if_exists()`. Without that, attempting to read the control will fail. We can assume the secondary controls exists since we require the TRUE based controls which also support the secondary. 
- The `||` should be broken up into two tests. 
- Drop the extra variables as they should not be needed. Also... check to make sure you are not defining `bool` anywhere else as it should be `auto`... otherwise your defining the type twice. 
- I would drop the using namespace here and use the full name as you should have more room once you move to two separate tests